discovery/gossiper.go

```diff
@@ -1579,6 +1680,64 @@ func (d *AuthenticatedGossiper) addNode(msg *lnwire.NodeAnnouncement,
 	return d.cfg.Router.AddNode(node, op...)
 }
 
+// isPremature decides whether a given network message has a block height+delta
+// value specified in the future. If so, the message will be added to the
+// future message map and be processed when the block height as reached.
+//
+// NOTE: must be used inside a lock.
+func (d *AuthenticatedGossiper) isPremature(chanID lnwire.ShortChannelID,
+	delta uint32, msg *networkMsg) bool {
+	// TODO(roasbeef) make height delta 6
+	//  * or configurable
+
+	msgHeight := chanID.BlockHeight + delta
+
+	// The message height is smaller or equal to our best known height,
+	// thus the message is mature.
+	if msgHeight <= d.bestHeight {
```

Comment:
Does this mean that if the msg is ahead by a few blocks (less than delta) it will be ignored as until now?

Reply:
I think the delta is 0. But if two nodes don't have the same view (e.g. one node gets the block quicker and sends channel_update), then this issue can happen in a real-world setting and not just itests

Reply:
For `ChannelAnnouncement` and `ChannelUpdate`, delta is 0. For `AnnounceSignatures` I think it's 6.